toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1477,45 +1479,97 @@ func ConfigureUserStartupCommand(installChroot *safechroot.Chroot, username stri
 	return
 }
 
-func ProvisionUserSSHCerts(installChroot *safechroot.Chroot, username string, sshPubKeyPaths []string) (err error) {
-	var (
-		pubKeyData []string
-		exists     bool
-	)
-	const squashErrors = false
-	const authorizedKeysTempFilePerms = 0644
-	const authorizedKeysTempFile = "/tmp/authorized_keys"
-	const sshDirectoryPermission = "0700"
+func setupSSHEnvironment(authorizedKeysTempFilePerms os.FileMode, authorizedKeysTempFile string) (err error) {
```

Comment:
If you place `setupSSHEnvironment` between `ProvisionUserSSHCertsWithPubKeys` and `ProvisionUserSSHCerts`, then the diff would be smaller.